src/LibraryManager.Vsix/Json/Completion/LibraryIdCompletionProvider.cs

```diff
@@ -81,7 +81,15 @@ protected override IEnumerable<JSONCompletionEntry> GetEntries(JSONCompletionCon
                     {
                         string insertionText = item.InsertionText.Replace("\\\\", "\\").Replace("\\", "\\\\");
                         ImageMoniker moniker = item.DisplayText.EndsWith("/") || item.DisplayText.EndsWith("\\") ? _folderIcon : _libraryIcon;
-                        yield return new SimpleCompletionEntry(item.DisplayText, insertionText, item.Description, moniker, trackingSpan, context.Session, ++count);
+
+                        if (set.Type == CompletionType.Version)
+                        {
+                            yield return new VersionCompletionEntry(item.DisplayText, insertionText, item.Description, moniker, trackingSpan, context.Session, ++count);
```

Comment:
This is used in SimpleCompletionEntry for sorting the entries. We want to keep the order of the completion entries as the provider sent to us.

---
In reply to: [198904839](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198904839) [](ancestors = 198904839)

Reply:
I add it to the VersionCompletionEntry here in case of the VersionCompletionEntry doesn't do the comparison (like empty version text VersionCompletionEntry), so that they can still be handled in SimpleCompletionEntry.

---
In reply to: [198933534](https://github.com/aspnet/LibraryManager/pull/258#discussion_r198933534) [](ancestors = 198933534,198904839)